### PR TITLE
add put(char) overload

### PR DESCRIPTION
**Pull request by WalterBright**
This trivial change reduces the size of the data section from 3126 to 2726 bytes, and the size of the object file from 26464 to 25041 bytes. There's a lot more that can be done to this module to reduce its footprint.

### CODE REVIEW

File: src/core/demangle.d [hunk 1]
```diff
@@ -223,6 +223,12 @@ private struct Demangle
     }
 
 
+    char[] put(char c)
+    {
+        char[1] val = c;
+        return put(val[]);
```
Comment (MartinNowak) [inline src/core/demangle.d]: You could simply use `put((&c)[0 .. 1])`.

Comment (WalterBright) [inline src/core/demangle.d]: true. Will do so next pass.

Comment (WalterBright) [inline src/core/demangle.d]: I decided against that because it makes the code not @safe

Comment (yebblies) [inline src/core/demangle.d]: It's not `@safe` either way, the compiler just lacks the check for slicing the static array.

**Pull request merged**